### ISSUE TITLE
[Bug]: Error when upgrading SB

### ISSUE THREAD

**Issue by anayap10** [storybookjs/storybook#25274]
### Describe the bug

I upgraded Storybook from 6 to 7. Some of the controls look like this:

<img width="538" alt="image" src="https://github.com/storybookjs/storybook/assets/82541208/13d90038-a1c0-483a-94ed-d099c752b9c0">

Besides, some of the Documentation throws me this error:

<img width="549" alt="image" src="https://github.com/storybookjs/storybook/assets/82541208/42bd2f1b-8c13-4bce-bb5a-4f3e2284f054">

This is my preview config

`export default {
  parameters: {
    actions: { argTypesRegex: '^on[A-Z].*' },
    controls: {
      matchers: {
        color: /(background|color)$/i,
        date: /Date$/i,
      },
    },
    docs: {
      toc: true, // 👈 Enables the table of contents
    },
  },
  decorators: [
    withThemeFromJSXProvider({
      themes: {
        light: lightTheme,
        dark: darkTheme,
      },
      defaultTheme: 'light',
      Provider: ThemeProvider,
    })]
};`

**Comment by apetta** [comment]
This seems to be coming from the controls add-on for me - disabling any controls on a story makes its docs work.

I think it might have something to do [this section in the docs](https://storybook.js.org/docs/configure/theming#addons-and-theme-creation) but it's not clear where we're supposed to set this.

**Comment by shilman** [comment]
Did you try removing `node_modules` and/or lockfiles and reinstalling?

**Comment by apetta** [comment]
> Did you try removing `node_modules` and/or lockfiles and reinstalling?

Yes to node modules, no to lock file yet as I'm in an RN project and it would cause a lot of chaos

**Comment by asfo** [comment]
I'm facing this issue also. I just updated my storybook from 6 to 7.

**Comment by xqwtsz** [comment]
> This seems to be coming from the controls add-on for me - disabling any controls on a story makes its doc work.

+1 also from me, with a tiny bit more extended error message

![Screenshot 2024-03-04 at 12 09 25](https://github.com/storybookjs/storybook/assets/3916893/91080091-9243-42a8-88a4-41d237381b31)

it looks like its caused by the `@storybook/addon-essentials`

removing/commenting out any boolean controls fixes it 🤔

**Comment by mbaumgart** [comment]
Apparently this is due to a version mismatch, I managed to fix it my explicitly adding the current version of `@storybook/theming` to my package json.

`npm i @storybook/theming`

**Comment by asfo** [comment]
> I'm facing this issue also. I just updated my storybook from 6 to 7.
> 
> My addons:
> 
> ```
> addons: [
>     "@storybook/addon-links",
>     '@storybook/addon-essentials',
>     "@storybook/addon-interactions",
>     "@storybook/preset-scss",
>     "@storybook/addon-mdx-gfm"
>   ],
> ```

Apparently, on our side, the problem was with the Angular templating, it auto-fixed it because a template internally had a problem with "BackgroundColor" which overrides the internal property so the new version doesn't work properly with Angular 14 but with Angular 16 works fine 🤔 and not sure why that happened or if it was a problem with theming + Angular or something.

Just in case this also occurs on your side, you need to upgrade to Angular 16+ and Storybook 7+ and that works better... (maybe with the issue above)
